record.go

```diff
@@ -34,6 +36,10 @@ func (c *Client) ReadRecord(domain string, hashId int) (*Record, error) {
 }
 
 func (c *Client) UpdateRecord(domain string, hashId int, record *Record) error {
+	if strings.EqualFold(record.RecordType, "CNAME") && !strings.HasSuffix(record.Address, ".") {
+		return fmt.Errorf("CNAME record to %s needs to end with a period '.'", record.Address)
```

Comment:
Can you please consider MX too? Corrupts state also, as mentioned in https://github.com/adamdecaf/terraform-provider-namecheap/issues/24

Reply:
Yea, I'm a bit rusty with this provider. Would comparing against `CNAME` or `MX` be enough here?

Reply:
(TL;DR at final paragraph haha)

By a soft reading of [this](https://stackexchange.github.io/dnscontrol/why-the-dot), and as a soft answer too, apparently the final period applies also for NS records, and for *anything where the “target” is a hostname*, BUT... appears that it is required a bit complicated validation, specifically, the idea is to look for ambiguities, rather than the presence/absence of the final period :thinking:

Taking the example there:

```
CNAME("foo", "bar")       // Permitted. (expands to bar.$DOMAIN)
CNAME("foo", "bar.com.")  // Permitted. (we are certain what the user wants)
CNAME("foo", "bar.com")   // ERROR (ambiguous)
CNAME("foo", "meta.xyz")  // ERROR (ambiguous)
```

The first example caught my attention, so I went directly to Namecheap, and appears that they don't allow that:

![image](https://user-images.githubusercontent.com/6588872/88621636-9697e500-d066-11ea-914c-db0e68c69fee.png)

But in the otherside, neither too haha. Adding a resource like...

```
resource "namecheap_record" "dom2" {
  domain = example.com
  name = "dom2"
  type = "CNAME"
  address = "dom"
  ttl = 1800
}
```

...had given me an output like:

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # namecheap_record.dom2 will be created
  + resource "namecheap_record" "dom2" {
      + address  = "dom"
      + domain   = "example.com"
      + hostname = (known after apply)
      + id       = (known after apply)
      + mx_pref  = 10
      + name     = "dom2"
      + ttl      = 1800
      + type     = "CNAME"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

namecheap_record.dom2: Creating...

Error: Failed to create namecheap Record: Number: 2050900 Message: Please enter a fully qualified domain name.


  on test.tf line 17, in resource "namecheap_record" "dom2":
  17: resource "namecheap_record" "dom2" {
```

But this time my state didn't get corrupted (yay)

So the final conclusion is that only the second example is valid: a FQDN finished by a period. Maybe adding a final period matching to this [FQDN validation regex](https://regexr.com/2rjcc) would be enough validation. (I know almost nothing about regex to add it, sorry :disappointed:)